coco-format-conversion/coco_format_conversion.py

```diff
@@ -0,0 +1,171 @@
+                ("area", area),
+                ("bbox",  [bb["@xmin"], bb["@ymin"], width, height]),
+                ("iscrowd", 0),
+            ])
+            (ret_annotations if i_title < len(titlelist) * train_val_cutoff
+             else ret_annotations_val if i_title < len(titlelist) *
+             val_test_cutoff else ret_annotations_test).append(annotation)
+
+        image_id_base += i_file + 1
+        annotation_id_base += i_id + 1
+    return ret_images, ret_annotations, ret_images_val, ret_annotations_val, ret_images_test, ret_annotations_test
+
+def getcoco(manga109_root_dir, train_val_cutoff, val_test_cutoff):
+    (ret_images, ret_annotations,
+     ret_images_val, ret_annotations_val,
+     ret_images_test, ret_annotations_test) = images_annotations(
+         manga109_root_dir, train_val_cutoff, val_test_cutoff)
+
+    data_training = OrderedDict([
+        ("info", d_info),
+        ("licenses", d_licenses),
+        ("images", ret_images),
+        ("annotations", ret_annotations),
+        ("categories", d_categories)
+    ])
+
+    data_val = OrderedDict([
+        ("info", d_info),
+        ("licenses", d_licenses),
+        ("images", ret_images_val),
+        ("annotations", ret_annotations_val),
+        ("categories", d_categories)
+    ])
+
+    data_test = OrderedDict([
+        ("info", d_info),
+        ("licenses", d_licenses),
+        ("images", ret_images_test),
+        ("annotations", ret_annotations_test),
+        ("categories", d_categories)
+    ])
+
+    return data_training, data_val, data_test
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Convert the Manga109 dataset XML to MSCOCO format.")
+    parser.add_argument("manga109_root_dir",
+                        type=str,
+                        help="The root directory of the Manga109 dataset.")
+    parser.add_argument("--outpath",
+                        type=pathlib.Path,
+                        default=pathlib.Path("./out/"),
+                        help="The output directory.")
+    parser.add_argument("--train-val-cutoff",
+                        type=float,
+                        default=0.8,
+                        help="Cutoff ratio between the training and the validation datasets.")
+    parser.add_argument("--val-test-cutoff",
+                        type=float,
+                        default=0.9,
+                        help="Cutoff ratio between the validation and the testing datasets.")
+    args = parser.parse_args()
+
+    manga109_root_dir = pathlib.Path(args.manga109_root_dir)
+
+    data_training, data_val, data_test = getcoco(manga109_root_dir, args.train_val_cutoff, args.val_test_cutoff)
+
+    args.outpath.mkdir(parents=True, exist_ok=True)
+
+    for outname, data_dict in zip(["train", "val", "test"], [data_training, data_val, data_test]):
+        coco_json = json.dumps(data_dict)
+        with open(str(args.outpath / "instances_{}.json".format(outname)), "wt") as f:
```

Comment:
Sorry, but I forget to comment last time.
You should you pathlib API like `(args.outpath / "instances_{}.json".format(outname)).open("wt")`.

Reply:
Thanks, I fixed it to use pathlib's API.